Modules/posixmodule.c

```diff
@@ -5477,13 +5498,17 @@ os__path_splitroot_ex_impl(PyObject *module, PyObject *path)
         goto exit;
     }
     tail = PyUnicode_FromWideChar(&buffer[drvsize + rootsize],
-                                  len - drvsize - rootsize);
+                                  path->length - drvsize - rootsize);
     if (tail == NULL) {
         goto exit;
     }
+    if (PyBytes_Check(path->object)) {
+        Py_SETREF(drv, PyUnicode_EncodeFSDefault(drv));
+        Py_SETREF(root, PyUnicode_EncodeFSDefault(root));
+        Py_SETREF(tail, PyUnicode_EncodeFSDefault(tail));
+    }
```

Comment:
We need to check `drv`, `root` and `tail` for `NULL` again after encoding in case it failed (or replace `PyTuple_Pack` with `Py_BuildValue("(NNN)", drv, root, tail)`, which handles incoming NULLs properly).

Reply:
I'll check manually, `Py_BuildValue()` is inefficient. Those null checks are very annoying though.